pwa-devdocs/src/tutorials/cloud-deploy/index.md

```diff
@@ -2,238 +2,198 @@
+magento-cloud checkout staging
 ```
 
-```sh
-composer require magento/module-sales-sample-data-venia:dev-master
-```
+## Add required Magento extensions
 
+Magento Cloud does not support node processes, so you cannot use UPWARD-JS to serve your storefront project.
+You must use the [magento2-upward-connector][] module with [UPWARD-PHP][] to deploy your storefront in Magento Cloud.
 
-These commands modify the `composer.json` file and adds the sample data modules to the `require` section:
+Use the `composer` CLI command to add the magento2-upward-connector module to the Magento installation:
 
-```json
-"magento/module-catalog-sample-data-venia": "dev-master",
-"magento/module-configurable-sample-data-venia": "dev-master",
-"magento/module-customer-sample-data-venia": "dev-master",
-"magento/module-tax-sample-data-venia": "dev-master",
-"magento/module-sales-sample-data-venia": "dev-master",
-"magento/sample-data-media-venia": "dev-master",
+```sh
+composer require magento/module-upward-connector
 ```
 
-## Install dependencies
-
-### Composer dependencies
-
-_Optional_: If you manually modified the `composer.json` file to make the changes described in the previous steps, update your `composer.lock` with these new dependencies:
+This command modifies the `composer.json` file and adds the package entry to the `require` section of the `composer.json` file.
 
-```sh
-composer update
+```text
+"magento/module-upward-connector": "^1.0.1"
 ```
 
-## Specify Cloud server environment
-
 {: .bs-callout .bs-callout-info}
-If you build your project locally and check the build artifacts into your Magento Cloud project, skip the **Install NPX and Yarn** and **Update build hooks** sections.
+UPWARD-PHP is a dependency of the magento2-upward-connector, so
+you do not need to add it manually to your project.
 
-### Install NPX and Yarn
+## Set environment variables
 
-If your project supports Yarn, which is the case for `venia-concept`, add the following entry to the `.magento.app.yaml` file.
+PWA Studio storefronts require you to set the following [environment variables][] in your Cloud project:
 
-```yaml
-dependencies:
-    nodejs:
-      yarn: "*"
-```
+| Name                                 | Description                                          |
+| ------------------------------------ | ---------------------------------------------------- |
+| `CONFIG__DEFAULT__WEB__UPWARD__PATH` | Absolute path to UPWARD YAML configuration           |
+| `NODE_ENV`                           | Specifies the node environment type                  |
+| `MAGENTO_BACKEND_URL`                | URL of your Magento backend                          |
+| `CHECKOUT_BRAINTREE_TOKEN`           | Braintree token associated with your Magento backend |
+| `MAGENTO_BACKEND_EDITION`            | Either `CE` or `EE`                                  |
+| `IMAGE_OPTIMIZING_ORIGIN`            | Origin to use for images in the UI                   |
```

Comment:
We may be long overdue for a doc about build time vs run time variables, but none of these should be needed in the cloud if you're checking pre-built assets into source.

Reply:
Addressed in 4ba1c88

Reply:
Sorry, my initial comment might not have been clear. We need all variables but `CONFIG__DEFAULT__WEB__UPWARD__PATH` at build time, and you need the first three in the table at runtime. Perhaps we can add columns to the table above to denote what's needed when, and describe that buildtime = what's needed in your `.env` and runtime = what's needed in `.magento.app.yaml` (or done in the Cloud UI).